drag_and_drop_v2/public/js/drag_and_drop.js

```diff
@@ -89,12 +99,30 @@ function DragNDropTemplates(url_name) {
             style['outline-color'] = item.color;
         }
         if (item.is_placed) {
-            style.left = item.x_percent + "%";
-            style.top = item.y_percent + "%";
-            if (item.widthPercent) {
+            if (item.zone_align === 'none') {
+                style.left = item.x_percent + "%";
+                style.top = item.y_percent + "%";
+            } else {
+                // Allow for the input + button width for aligned items
+                if (item.input) {
+                    style.marginRight = '190px';
+                }
+            }
+            if (item.zone_align !== 'none') {
```

Comment:
@itsjeyd Good point. In earlier iterations there wasn't a clean way to combine these, but with the way I separated out the image-size handling at the end, these should now be combined.